modules/tinymce/src/plugins/table/test/ts/browser/ClipboardTest.ts

```diff
@@ -421,6 +421,182 @@ UnitTest.asynctest('browser.tinymce.plugins.table.ClipboardTest', (success, fail
+    editor.execCommand('mceTablePasteColAfter');
+
+    LegacyUnit.equal(
+      cleanTableHtml(editor.getContent()),
+      '<table>' +
+      '<tbody>' +
+      '<tr><td colspan="2">1</td><td>3</td><td colspan="2">1</td><td>3</td></tr>' +
+      '<tr><td>1</td><td colspan="2">2</td><td>1</td><td colspan="2">2</td></tr>' +
+      '<tr><td>1</td><td>2</td><td>3</td><td>1</td><td>2</td><td>3</td></tr>' +
+      '</tbody>' +
+      '</table>'
+    );
+  });
+
+  suite.test('TestCase-TINY-6006: Table: col clipboard api', (editor) => {
+    editor.setContent(
+      '<table>' +
+      '<tr><td>1</td><td>2</td></tr>' +
+      '<tr><td>2</td><td>3</td></tr>' +
+      '</table>'
+    );
+
+    LegacyUnit.setSelection(editor, 'tr td:nth-child(1)', 0);
+    editor.execCommand('mceTableCopyCol');
+
+    const clipboardCols = editor.plugins.table.getClipboardCols();
+
+    LegacyUnit.equal(clipboardCols.length, 2);
+    LegacyUnit.equal(clipboardCols[0].tagName, 'TR');
+    const cells = clipboardCols[0].childNodes;
+    LegacyUnit.equal(cells.length, 1);
+    LegacyUnit.equal(cells[0].nodeName, 'TD');
+
+    editor.plugins.table.setClipboardCols([
+      createRow(editor, [ 'a', 'b' ]),
+      createRow(editor, [ 'c', 'd' ])
+    ]);
+
+    LegacyUnit.setSelection(editor, 'tr td:nth-child(2)', 0);
+    editor.execCommand('mceTablePasteColAfter');
+
+    LegacyUnit.equal(
+      cleanTableHtml(editor.getContent()),
+
+      '<table>' +
+      '<tbody>' +
+      '<tr><td>1</td><td>2</td><td>a</td><td>b</td></tr>' +
+      '<tr><td>2</td><td>3</td><td>c</td><td>d</td></tr>' +
+      '</tbody>' +
+      '</table>'
+    );
+  });
+
+  suite.test('TestCase-TINY-6684: Table: mceTablePasteColBefore command with colgroups', (editor) => {
+    editor.setContent(
+      '<table>' +
+      '<colgroup><col><col></colgroup>' +
+      '<tbody>' +
+      '<tr><td>1</td><td>2</td></tr>' +
+      '<tr><td>2</td><td>3</td></tr>' +
+      '</tbody>' +
+      '</table>'
+    );
+
+    selectOne(editor, 'tr td:nth-child(2)');
+    editor.execCommand('mceTableCopyCol');
+    selectOne(editor, 'tr td:nth-child(1)');
+    editor.execCommand('mceTablePasteColBefore');
+
+    LegacyUnit.equal(
+      cleanTableHtml(editor.getContent()),
+
+      '<table>' +
+      '<colgroup><col /><col /><col /></colgroup>' +
+      '<tbody>' +
+      '<tr><td>2</td><td>1</td><td>2</td></tr>' +
+      '<tr><td>3</td><td>2</td><td>3</td></tr>' +
+      '</tbody>' +
+      '</table>'
+    );
+  });
+
+  suite.test('TestCase-TINY-6684: Table: mceTablePasteColAfter command with colgroups', (editor) => {
+    editor.setContent(
+      '<table>' +
+      '<colgroup><col><col></colgroup>' +
+      '<tbody>' +
+      '<tr><td>1</td><td>2</td></tr>' +
+      '<tr><td>2</td><td>3</td></tr>' +
+      '</tbody>' +
+      '</table>'
+    );
+
+    selectOne(editor, 'tr td:nth-child(1)');
+    editor.execCommand('mceTableCopyCol');
+    selectOne(editor, 'tr td:nth-child(1)');
```

Comment:
Would it be better to select the second column so it is more clear the col is inserted after in the HTML?

Reply:
That's already being done in a previous test. I wanted to have this being a little different, so as more cases are covered via automation.